### ISSUE TITLE
Unable to sync App Service - deployment failure

### ISSUE THREAD

**Issue by pgriziotti** [cristianoag/microsoft-teams-apps-company-communicator#20]
The App service fail to sync.
The log reports:

Command: deploy.cmd
Handling ASP.NET Core Web Application deployment.
Restoring NuGet packages
  Determining projects to restore...
  All projects are up-to-date for restore.
Restoring npm packages (this can take several minutes)
npm WARN @babel/plugin-bugfix-v8-spread-parameters-in-optional-chaining@7.17.12 requires a peer of @babel/core@^7.13.0 but none is installed. You must install peer dependencies yourself.
up to date in 151.027s
npm WARN @pmmmwh/react-refresh-webpack-plugin@0.4.3 requires a peer of type-fest@^0.13.1 but none is installed. You must install peer dependencies yourself.
npm WARN @pmmmwh/react-refresh-webpack-plugin@0.4.3 requires a peer of webpack-hot-middleware@2.x but none is installed. You must install peer dependencies yourself.
Building the client app (this can take several minutes)
npm WARN @pmmmwh/react-refresh-webpack-plugin@0.4.3 requires a peer of webpack-plugin-serve@0.x || 1.x but none is installed. You must install peer dependencies yourself.
npm WARN jest-config@26.6.3 requires a peer of ts-node@>=9.0.0 but none is installed. You must install peer dependencies yourself.
npm WARN jsdom@16.7.0 requires a peer of canvas@^2.5.0 but none is installed. You must install peer dependencies yourself.
npm WARN node-fetch@2.6.7 requires a peer of encoding@^0.1.0 but none is installed. You must install peer dependencies yourself.
npm WARN sass-loader@10.2.1 requires a peer of fibers@>= 3.1.0 but none is installed. You must install peer dependencies yourself.
npm WARN sass-loader@10.2.1 requires a peer of node-sass@^4.0.0 || ^5.0.0 || ^6.0.0 but none is installed. You must install peer dependencies yourself.
npm WARN ws@7.5.7 requires a peer of bufferutil@^4.0.1 but none is installed. You must install peer dependencies yourself.
npm WARN ws@7.5.7 requires a peer of utf-8-validate@^5.0.2 but none is installed. You must install peer dependencies yourself.
npm WARN The package papaparse is included as both a dev and production dependency.
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.13 (node_modules\watchpack-chokidar2\node_modules\fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.13: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"ia32"})

> company-communicator@4.1.5 build C:\home\site\repository\Source\CompanyCommunicator\ClientApp
> react-scripts build 

Creating an optimized production build...
Compiled with warnings.

src\components\ManageGroups\ManageGroups.tsx
  Line 1:1:  Unexpected Unicode BOM (Byte Order Mark)  unicode-bom

src\components\NewMessage\newMessage.tsx
  Line 35:7:      'coeff' is assigned a value but never used                @typescript-eslint/no-unused-vars
  Line 488:15:    'buttonTitleAsString' is assigned a value but never used  @typescript-eslint/no-unused-vars
  Line 740:119:   Expected '===' and instead saw '=='                       eqeqeq
  Line 1597:139:  Expected '===' and instead saw '=='                       eqeqeq
  Line 1630:162:  Expected '===' and instead saw '=='                       eqeqeq
  Line 1671:162:  Expected '===' and instead saw '=='                       eqeqeq

src\components\OverFlow\scheduledMessageOverflow.tsx
  Line 1:1:  Unexpected Unicode BOM (Byte Order Mark)  unicode-bom

Search for the keywords to learn more about each warning.
To ignore, add // eslint-disable-next-line to the line before.

File sizes after gzip:

  468.36 KB  build\static\js\2.24637181.chunk.js
  17.02 KB   build\static\js\main.b18cdb9f.chunk.js
  2.97 KB    build\static\css\main.93868222.chunk.css
  783 B      build\static\js\runtime-main.f341037b.js

The project was built assuming it is hosted at /.
You can control this with the homepage field in your package.json.

The build folder is ready to be deployed.
You may serve it with a static server:

  npm install -g serve
  serve -s build

Find out more about deployment here:

  https://cra.link/deployment

Building the application
Microsoft (R) Build Engine version 16.7.2+b60ddb6f4 for .NET
Copyright (C) Microsoft Corporation. All rights reserved.

  Determining projects to restore...
  All projects are up-to-date for restore.
  Microsoft.Teams.Apps.CompanyCommunicator.Common -> C:\home\site\repository\Source\CompanyCommunicator.Common\bin\Release\netstandard2.1\Microsoft.Teams.Apps.CompanyCommunicator.Common.dll
  Microsoft.Teams.Apps.CompanyCommunicator -> C:\home\site\repository\Source\CompanyCommunicator\bin\Release\netcoreapp3.1\Microsoft.Teams.Apps.CompanyCommunicator.dll
  Microsoft.Teams.Apps.CompanyCommunicator -> C:\home\site\repository\Source\CompanyCommunicator\bin\Release\netcoreapp3.1\Microsoft.Teams.Apps.CompanyCommunicator.Views.dll
  Restoring dependencies using 'npm'. This may take several minutes...
  Building the client app using 'npm'. This may take several minutes...
  Finished building the client app
  Microsoft.Teams.Apps.CompanyCommunicator -> C:\local\Temp\8da3bc0a4cf5bb3\
Creating app_offline.htm
KuduSync.NET from: 'C:\local\Temp\8da3bc0a4cf5bb3' to: 'C:\home\site\wwwroot'
Copying file: 'Microsoft.Teams.Apps.CompanyCommunicator.deps.json'
Copying file: 'ClientApp\build\asset-manifest.json'
Copying file: 'ClientApp\build\index.html'
Copying file: 'ClientApp\build\static\css\main.93868222.chunk.css'
Copying file: 'ClientApp\build\static\css\main.93868222.chunk.css.map'
Copying file: 'ClientApp\build\static\js\2.24637181.chunk.js'
Copying file: 'ClientApp\build\static\js\2.24637181.chunk.js.LICENSE.txt'
Copying file: 'ClientApp\build\static\js\2.24637181.chunk.js.map'
Copying file: 'ClientApp\build\static\js\main.b18cdb9f.chunk.js'
Copying file: 'ClientApp\build\static\js\main.b18cdb9f.chunk.js.map'
Copying file: 'ClientApp\build\static\js\runtime-main.f341037b.js'
Copying file: 'ClientApp\build\static\js\runtime-main.f341037b.js.map'
Copying file: 'ClientApp\build\static\media\getFetch.582d29c4.cjs'
Copying file: 'ClientApp\build\static\media\TeamsAssets-Light.14c3e6f0.eot'
Copying file: 'ClientApp\build\static\media\TeamsAssets-Light.66561737.ttf'
Copying file: 'ClientApp\build\static\media\TeamsAssets-Light.6d0fa9a8.woff'
Copying file: 'ClientApp\build\static\media\TeamsAssets-Light.73e6fac8.svg'
Copying file: 'ClientApp\build\static\media\TeamsAssets-Light.e8bd73e2.woff2'
Copying file: 'ClientApp\build\static\media\TeamsAssets-Regular.59b42e51.woff2'
Copying file: 'ClientApp\build\static\media\TeamsAssets-Regular.5da5870e.eot'
Copying file: 'ClientApp\build\static\media\TeamsAssets-Regular.7239f6ef.svg'
Copying file: 'ClientApp\build\static\media\TeamsAssets-Regular.a49d84c5.woff'
Copying file: 'ClientApp\build\static\media\TeamsAssets-Regular.ace6ce92.ttf'
Deleting app_offline.htm
Finished successfully.
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.13: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"ia32"})
Input string was not in a correct format.

**Comment by ahmedmahmoudR** [comment]
I have  same issue

**Comment by TerryLawSC** [comment]
Same issue here.

**Comment by pgriziotti** [comment]
Sounds working for me @cristianoag 
The npm command in the console reports:

found 16 vulnerabilities (2 low, 5 moderate, 8 high, 1 critical)
  run `npm audit fix` to fix them, or `npm audit` for details

Needs to be fixed?
Thanks!

**Comment by cristianoag** [comment]
Have you managed to finish the sync after issuing the command? I do think we need to update the npms to fix those vulnerabilities, but first make sure the app service syncs correctly.

**Comment by pgriziotti** [comment]
Yes @cristianoag the sync completes successfully, and the Communicator works!
Thanks!
